src/fidasim.f90

```diff
@@ -7432,19 +7463,29 @@ subroutine halo
                         !! Weight CX rates by ion source density
                         states = rates*plasma%denp
 
+                        allocate(idx_arr(3,ncell),denn_arr(nlevs,ncell))
                         loop_along_track: do jj=1,ncell
                             call get_plasma(plasma,pos=tracks(jj)%pos)
 
                             call colrad(plasma,thermal_ion,vihalo,tracks(jj)%time,states,denn,photons)
-                            call store_neutrals(tracks(jj)%ind,s2type, &
-                                 denn/nlaunch(i,j,k),vihalo,plasma%in_plasma)
+                            idx_arr(:,jj) = tracks(jj)%ind
+                            denn_arr(:,jj) = denn*r_nlaunchijk
+                            if (plasma%in_plasma) then
+                               local_dens = local_dens + sum(denn_arr(:,jj))
+                            endif
 
                             if((photons.gt.0.d0).and.(inputs%calc_bes.ge.1)) then
-                              call store_bes_photons(tracks(jj)%pos,vihalo,photons/nlaunch(i,j,k),halo_type)
+                              call store_bes_photons(tracks(jj)%pos,vihalo,photons*r_nlaunchijk,halo_type)
                             endif
 
                         enddo loop_along_track
+
+                        call store_neutrals_multi(ncell,idx_arr,s2type,denn_arr)
```

Comment:
This sort of trick could also be done in the `dcx` and `ndmc` subroutines.

Reply:
True, but the contention there is minimal. 
Not worth it.